internal/controller/proxy.go

```diff
@@ -0,0 +1,22 @@
+package controller
+
+import (
+	"net/http"
+	"net/url"
+	"os"
+
+	"go.uber.org/zap"
+)
+
+func configureProxy(logger *zap.Logger) func(*http.Request) (*url.URL, error) {
```

Comment:
Why did you need to reimplement the standard `http.ProxyFromEnvironment`?
Explain that in a comment for this function.